mr/src/main/java/org/apache/iceberg/mr/mapred/TableResolver.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.mr.mapred;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Properties;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.mapred.JobConf;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.hadoop.HadoopCatalog;
+import org.apache.iceberg.hadoop.HadoopTables;
+import org.apache.iceberg.mr.InputFormatConfig;
+
+final class TableResolver {
```

Comment:
I don't see the need to for having two table resolver classes. Can we combine them? The only difference I see is that this class uses `org.apache.hadoop.mapred.JobConf` instead of `org.apache.hadoop.mapreduce.JobConf`. It seems like we don't use any `JobConf` methods, and if we have to, we can create 2 methods instead of 2 classes.

Reply:
The intention was to not have a class that has both the `mapred` and the `mapreduce` apis as imports just to keep the APIs completely separate and prevent mistakes. If we feel it would be beneficial to combine them into one and it's still simple to see which paths are used by which APIs (where they don't overlap) then I guess we could combine them. As it stands right now I'm not sure there is enough overlapping code to justify doing this?

Reply:
👍  I think the overall functionality offered by both classes is the same. Although there is some feature parity at the moment w.r.t. metadata tables. I guess we can revisit this at a later point.